### DISCUSSION TITLE
Used to compute Hausdorff distances between polytopes

### DISCUSSION THREAD

**Discussion by geodavic** [qpsolvers/qpsolvers#69] [Show and tell]
I found this tool useful in my research, which required me to compute the [Hausdorff distance](https://en.wikipedia.org/wiki/Hausdorff_distance) between two polytopes. If using the standard Euclidean metric, this can be formulated using a quadratic program. I had to cycle through several solvers in order to get it to work, so having the unified API was extremely helpful! The one that ended up working was `osqp`. I have attached an example.

![Screen Shot 2022-07-14 at 10 39 14 AM](https://user-images.githubusercontent.com/41129492/179022458-14785fea-d613-44ec-973f-6db43d4b6b70.png)

I'll make sure to credit this repo in my paper!

**Comment by stephane-caron** [comment]
Very nice :smiley: Feel free to share links to your paper/code once it is out.

**Reply by geodavic** [reply]
@stephane-caron my paper and code are finally out!

Paper: https://arxiv.org/abs/2407.13125
Code: https://github.com/geodavic/zonopt

**Comment by ogencoglu** [comment]
@geodavic  can you link the snippet to perform the hausdorff distance computation between arbitrary polygons using `osqp`? Or is it so that this formulation can only be done for polygon-zonotype distance computation, instead of 2 arbitrary polygons?

**Reply by geodavic** [reply]
@ogencoglu Sure -- that is performed here: https://github.com/geodavic/zonopt/blob/758b14d0a10f1b3344ecdf97ad6b9a1f8eb89602/zonopt/metrics/hausdorff.py#L200

It is on arbitrary polytopes.
